packages/vite/src/node/plugins/css.ts

```diff
@@ -2083,16 +2092,22 @@ async function doImportCSSReplace(
 ) {
   let wrap = ''
   const first = rawUrl[0]
+  let unquotedUrl = rawUrl
   if (first === `"` || first === `'`) {
     wrap = first
-    rawUrl = rawUrl.slice(1, -1)
+    unquotedUrl = rawUrl.slice(1, -1)
+  }
+  if (skipUrlReplacer(unquotedUrl)) {
+    return matched
   }
-  if (isExternalUrl(rawUrl) || isDataUrl(rawUrl) || rawUrl[0] === '#') {
```

Comment:
This now checks `functionCallRE.test(unquotedUrl)` additionally.
The following code works and the URL replacement should be skipped.
```scss
@function foo() {
  @return 'foo.css';
}

@import url(foo());
```